### ISSUE TITLE
Won't compile: 'BearSSLPublicKey' was not declared in this scope

### ISSUE THREAD

**Issue by kaderud** [pushdata-io/Arduino_ESP8266_SSL#1]
Why is line 123 commented out:
`//BearSSL::PublicKey key;`

and instead on line 122 this is used:
`BearSSLPublicKey key;`

With `BearSSLPublicKey` I get compiler error:

Pushdata_ESP8266_SSL/src/Pushdata_ESP8266_SSL.h:123:13: error: 'BearSSLPublicKey' was not declared in this scope

But not with `BearSSL::PublicKey key;`

**Comment by kaderud** [comment]
Using Arduino 1.8.8 under Linux; tested with NodeMCU 1.0, using ESP8266 Core v2.5.0

**Comment by ragnarlonn** [comment]
@kaderud This is kind of a thorny issue maybe. Are you using the Arduino IDE?  

I have had issues with the BearSSL namespace when developing in the PlatformIO IDE - I think this is the underlying problem:

- BearSSL used to declare their classes in the global namespace, but started using the namespace "BearSSL::" quite recently
- The ESP8266WiFi library bundles BearSSL code, and seem to have updated the bundled code so it uses the new namespace, but they have not updated the version number for ESP8266WiFi
- PlatformIO will not update ESP8266WiFi unless the version number is bumped in the public Github repo (https://github.com/esp8266/Arduino/tree/master/libraries/ESP8266WiFi)

I have been developing on a couple of different machines, and at one point found that I had to use "BearSSL::PublicKey" to declare a public key object, then later on (when on a vanilla PlatformIO setup) that wouldn't work anymore, so then I changed it to the older "BearSSLPublicKey" from the global namespace. This will of course not work on a system that uses a newer ESP8266WiFi library, that bundles newer BearSSL code (while still being version 1.0)

Not sure how to solve this in the best way, so it bites as few people as possible. Would be nice if ESP8266WiFi started updating their version numbers, of course, but in case they won't, a workaround would be nice. Hmm, maybe there is some #ifdef I can use that detects if the user is using PlatformIO?

**Comment by kaderud** [comment]
> @kaderud This is kind of a thorny issue maybe. Are you using the Arduino IDE?

It's not my preferred choice :-) but in this case, since it was quickest and easiest, I did my tests in Arduino IDE 1.8.8 under Linux, but my preference is Visual Studio Code, I'll try and setup a workspace for that and give it a try. I might have Atom/PlatformIO installed but haven't used it in ages.

I'm in no way a C++ coder, my level is basically `cout` / `cin`, but I can make my way around C source, so I'll do some more digging and see if I can come up with some more information to help with a possible `#ifdef` for this issue!

**Comment by ragnarlonn** [comment]
I use VS code too (love that editor) - and PlatformIO is available as a plugin for VS code!  Not the best integration I've seen, UI wise (I think it's slightly unintuitive which controls belong to PlatformIO and which are VS code native controls) but the functionality is very nice, with a decent library manager and one-click library installs, compilation & upload and a serial monitor. It is like the Arduino IDE but better in every way (IMHO).

**Comment by pushdata-io** [comment]
@kaderud I just tested reinstalling PlatformIO, and my issue with the "BearSSL::PublicKey" declaration disappeared. Then I tested that declaration in the Arduino IDE also, and it worked fine, so it seems my problems may have been to a corrupt PlatformIO installation.

I changed the line now, and have released a 0.0.8 version of the library so if you update it, it should work for you.

Thanks a lot for helping out with this. I'm new to writing Arduino libraries (and fairly new to Arduino development in general).
